STM32F7 ARM scatter file update

jeromecoutant opened the pull request:
### Description

This fix #10313 

Update done:
- use predefine values from pack manager
- define the 2 RAM
- define the crash data block for all targets

Next:
- align with other STM32F7 targets
- check other toolchains


### Pull request type

    [x] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

jeromecoutant committed:
STM32F7 ARM scatter file

mprse reviewed (approved):
I suggest to add:
```
#if !defined(MBED_BOOT_STACK_SIZE)	
  #define MBED_BOOT_STACK_SIZE 0x400	
#endif
```

This is now the common pattern in all liker scripts.

Otherwise ok.

jeromecoutant commented on targets/TARGET_STM/TARGET_STM32F7/TARGET_STM32F746xG/device/TOOLCHAIN_ARM_STD/stm32f746xg.sct:
```diff
@@ -28,30 +28,28 @@
+; MBED_RAM_START =0x20010000 MBED_RAM_SIZE =0x40000 (256 KB)
+; MBED_RAM1_START=0x20000000 MBED_RAM1_SIZE=0x10000 (64 KB)
+
+#include "../cmsis_nvic.h"
```
This include seems not compatible with MBED OS2...

jeromecoutant committed:
STM32F7 ARM scatter file update

- startup file: MBED_RAM_START and MBED_RAM_SIZE not known in non mbed-cli env
- include file can not be called in MBED-OS2 env
- align ARM_STD and ARM_MICRO

deepikabhavnani commented on targets/TARGET_STM/TARGET_STM32F7/TARGET_STM32F746xG/device/TOOLCHAIN_ARM_MICRO/stm32f746xg.sct:
```diff
@@ -28,55 +28,46 @@
-#define MBED_IRAM1_START             (MBED_RAM_START + VECTOR_SIZE + MBED_CRASH_REPORT_RAM_SIZE)
-#define MBED_IRAM1_SIZE              (MBED_RAM_SIZE - VECTOR_SIZE - MBED_CRASH_REPORT_RAM_SIZE)
+#define MBED_RAM0_START             (MBED_CRASH_REPORT_RAM_START + MBED_CRASH_REPORT_RAM_SIZE)
+#define MBED_RAM0_SIZE              (MBED_RAM1_SIZE + MBED_RAM_SIZE - MBED_VECTTABLE_RAM_SIZE - MBED_CRASH_REPORT_RAM_SIZE)
```
```suggestion
#define MBED_RAM0_SIZE              (MBED_RAM1_SIZE - MBED_VECTTABLE_RAM_SIZE - MBED_CRASH_REPORT_RAM_SIZE)
```

deepikabhavnani replied:
RAM1 and RAM0 are physically separate regions, I am not sure if we can use them as contiguous single region, was this tested in various scenarios.

As per my understanding here below is the memory map:

![image](https://user-images.githubusercontent.com/28542450/55835127-fddffd80-5ae0-11e9-843f-a9cf6dc03b16.png)

Will like to know if case 1 and 2 is tested somehow, 
1. RW_IRAM1 if ends before RAM_START, heap access across RAM and RAM1 is successful 
2. RW_IARM1 spans across RAM boundary

jeromecoutant replied:
Hi

I don't agree with your suggestion...
Solution could be to start RAM0 at RAM?

Information from STM32F746 Reference Manual:
- SRAM1    mapped at 0x20010000 size 0x3C000
- SRAM2    mapped at 0x2004C000 size  0x4000
- DTCM-RAM mapped at 0x20000000 size 0x10000

Information from STM32F7xx pdsc file:
- IRAM1    mapped at 0x20010000 size 0x40000 ==> RAM in mbed-os pack manager
- IRAM2    mapped at 0x20000000 size 0x10000 ==> RAM1 in mbed-os pack manager
